File: client_sdks/devconnect-android/src/main/java/com/devconnect/reporters/LogReporter.kt
package com.devconnect.reporters

import com.devconnect.DevConnect

/**
 * Tagged logger for DevConnect.
 *
 * ```kotlin
 * val logger = DevConnect.logger("AuthService")
 * logger.info("User logged in")
 * logger.error("Login failed", stackTrace = Log.getStackTraceString(e))
 * ```
 */
class LogReporter(private val tag: String? = null) {

    fun debug(message: String, metadata: Map<String, Any>? = null) {
        DevConnect.debug(message, tag, metadata)
    }

    fun info(message: String, metadata: Map<String, Any>? = null) {
        DevConnect.log(message, tag, metadata)
    }

    fun warn(message: String, metadata: Map<String, Any>? = null) {
        DevConnect.warn(message, tag, metadata)
    }

    fun error(
        message: String,
        stackTrace: String? = null,
        metadata: Map<String, Any>? = null
    ) {
        DevConnect.error(message, tag, stackTrace, metadata)
    }

    /**
     * Log an exception with full stack trace.
     */
    fun exception(e: Throwable, message: String? = null) {
        DevConnect.error(
            message = message ?: e.message ?: e.toString(),
            tag = tag,
            stackTrace = e.stackTraceToString()
        )
    }
}

File: client_sdks/devconnect_flutter/lib/src/reporters/log_reporter.dart
import '../devconnect_client.dart';

/// Convenience class for logging to DevConnect.
///
/// Usage:
/// ```dart
/// final logger = DevConnectLogger(tag: 'AuthService');
/// logger.info('User logged in');
/// logger.error('Login failed', stackTrace: StackTrace.current.toString());
/// ```
class DevConnectLogger {
  final String? tag;

  const DevConnectLogger({this.tag});

  void debug(String message, {Map<String, dynamic>? metadata}) {
    DevConnectClient.safeSendLog(level: 'debug', message: message, tag: tag, metadata: metadata);
  }

  void info(String message, {Map<String, dynamic>? metadata}) {
    DevConnectClient.safeLog(message, tag: tag, metadata: metadata);
  }

  void warn(String message, {Map<String, dynamic>? metadata}) {
    DevConnectClient.safeSendLog(level: 'warn', message: message, tag: tag, metadata: metadata);
  }

  void error(String message,
      {String? stackTrace, Map<String, dynamic>? metadata}) {
    DevConnectClient.safeSendLog(level: 'error', message: message, tag: tag, stackTrace: stackTrace, metadata: metadata);
  }
}
